src/parcsr_ls/ams.c

```diff
@@ -8,6 +8,7 @@
 #include "_hypre_parcsr_ls.h"
 #include "float.h"
 #include "ams.h"
+#include "hypre_cuda_utils.h"
```

Comment:
I suppose I can check myself, but I was wondering how many files that are not '*_device.c' files have this include?

Reply:
I don't like the fact that this file needed to be compiled by `CUCC`. The problem is due to some junk such as this line https://github.com/hypre-space/hypre/blob/3f8ff325b4577e055d8f826eb2a88d52c0fe7b23/src/parcsr_ls/ams.c#L697, which needs CUDA and so the headers. It is an example of not isolating our functions properly. What we can do is to make a function for that particular operation, and hide the actual implementation there, so `ams.c` does not see it and should not care about it at all.